src/routes/devices.ts

```diff
@@ -804,169 +786,159 @@ export const statePatch: RequestHandler = (req, res) => {
 		custom.ipAddress = getIP(req);
 	}
 
-	return sbvrUtils.db
-		.transaction(tx => {
+	try {
+		await sbvrUtils.db.transaction(async tx => {
 			const resinApiTx = api.resin.clone({ passthrough: { req, custom, tx } });
 
-			return resinApiTx
-				.get({
-					resource: 'device',
+			const [device] = (await resinApiTx.get({
+				resource: 'device',
+				options: {
+					$select: 'id',
+					$filter: { uuid },
+				},
+			})) as AnyObject[];
+
+			if (device == null) {
+				throw new UnauthorizedError();
+			}
+
+			if (local.is_on__commit === null) {
+				deviceBody!.is_running__release = null;
```

Comment:
I was hoping to get rid of this `!` and the equivalent one below, but turns out it's still required because the check is outside of the `sbvrUtils.db.transaction` and this is in the callback :(